What percent of non first class passengers survived?

I will be using an retrospective observational study to answer my question, as I will be going through the dataset that I have to try and answer this question using my census sampling.

In [46]:
# Import pandas
import pandas as pd
import numpy as np

# Verify pandas is imported correctly by checking the version
print(f"Pandas version: {pd.__version__}")

# You can now use pandas in your notebook

Pandas version: 2.1.4


In [47]:
# Load a CSV file into a DataFrame
df = pd.read_csv('titanic.csv')
df.info

<bound method DataFrame.info of      PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                   

In [48]:
# Method 1: Using the columns attribute to display the collumns from your dataframe.
print("Displaying collumns using df.columns")
print(df.columns)

Displaying collumns using df.columns
Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')


In [49]:
# Method 1: Using len() function to determine how many samples are found in my dataset.
count_1 = len(df)
print(f"Method 1 - Using len(): {count_1} samples")

Method 1 - Using len(): 891 samples


In [50]:
# Method 1: Accessing the seventh sample using iloc (integer-location based indexing)
# This is the most straightforward way to access a row by position
seventh_sample_iloc = df.iloc[6]
print("Method 1 - Using df.iloc[6]:")
print(seventh_sample_iloc)
print("\n" + "-"*50 + "\n")

Method 1 - Using df.iloc[6]:
PassengerId                          7
Survived                             0
Pclass                               1
Name           McCarthy, Mr. Timothy J
Sex                               male
Age                               54.0
SibSp                                0
Parch                                0
Ticket                           17463
Fare                           51.8625
Cabin                              E46
Embarked                             S
Name: 6, dtype: object

--------------------------------------------------



In [51]:
df[df["Embarked"] == "Q"]

PassengerId  Survived  Pclass                                  Name  \
5              6         0       3                      Moran, Mr. James   
16            17         0       3                  Rice, Master. Eugene   
22            23         1       3           McGowan, Miss. Anna "Annie"   
28            29         1       3         O'Dwyer, Miss. Ellen "Nellie"   
32            33         1       3              Glynn, Miss. Mary Agatha   
..           ...       ...     ...                                   ...   
790          791         0       3              Keane, Mr. Andrew "Andy"   
825          826         0       3                       Flynn, Mr. John   
828          829         1       3          McCormack, Mr. Thomas Joseph   
885          886         0       3  Rice, Mrs. William (Margaret Norton)   
890          891         0       3                   Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
5      male   NaN      0      0  330877   8.4583   NaN        Q  
16     male   2.0      4      1  382652  29.1250   NaN        Q  
22   female  15.0      0      0  330923   8.0292   NaN        Q  
28   female   NaN      0      0  330959   7.8792   NaN        Q  
32   female   NaN      0      0  335677   7.7500   NaN        Q  
..      ...   ...    ...    ...     ...      ...   ...      ...  
790    male   NaN      0      0   12460   7.7500   NaN        Q  
825    male   NaN      0      0  368323   6.9500   NaN        Q  
828    male   NaN      0      0  367228   7.7500   NaN        Q  
885  female  39.0      0      5  382652  29.1250   NaN        Q  
890    male  32.0      0      0  370376   7.7500   NaN        Q  

[77 rows x 12 columns]

In [52]:
# Load the Titanic dataset
try:
    # Try to load from a local file first
    titanic = pd.read_csv('titanic.csv')
except:
    # If that fails, load the sample Titanic dataset
    import seaborn as sns
    titanic = sns.load_dataset('titanic')

In [53]:
# Display the first few rows to understand the data structure
print("First few rows of the dataset:")
print(titanic.head())

First few rows of the dataset:
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            3734

In [54]:
# Check column names to confirm we have 'survived' and 'pclass' columns
print("\nColumn names:", list(titanic.columns))


Column names: ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


In [55]:
# Calculate the number of non-first class survivors
# First class is represented as pclass=1
non_first_class_survivors = titanic[(titanic['Survived'] == 1) & (titanic['Pclass'] != 1)]

In [56]:
# Calculate the percentage relative to all passengers
total_passengers = len(titanic)
percentage_of_all = (len(non_first_class_survivors) / total_passengers) * 100

In [57]:
# Calculate the percentage relative to all non-first class passengers
non_first_class_passengers = titanic[titanic['Pclass'] != 1]
percentage_of_non_first = (len(non_first_class_survivors) / len(non_first_class_passengers)) * 100

In [58]:
# Calculate the percentage relative to all survivors
all_survivors = titanic[titanic['Survived'] == 1]
percentage_of_survivors = (len(non_first_class_survivors) / len(all_survivors)) * 100

In [59]:
# Display the results
print(f"\nTotal passengers: {total_passengers}")
print(f"Non-first class survivors: {len(non_first_class_survivors)}")
print(f"Percentage of all passengers who were non-first class survivors: {percentage_of_all:.2f}%")
print(f"Percentage of non-first class passengers who survived: {percentage_of_non_first:.2f}%")
print(f"Percentage of survivors who were non-first class: {percentage_of_survivors:.2f}%")


Total passengers: 891
Non-first class survivors: 206
Percentage of all passengers who were non-first class survivors: 23.12%
Percentage of non-first class passengers who survived: 30.52%
Percentage of survivors who were non-first class: 60.23%


Only 30.52% of all the non-first class passengers survived the sinking of the titanic.

My study may contain biases in information because it is incomplete, meaning it may over or underrepresent certain groups, and 